core/templates/pages/exploration-editor-page/editor-tab/services/responses.service.ts

```diff
@@ -323,10 +323,60 @@ angular.module('oppia').factory('ResponsesService', [
       // Updates answer choices when the interaction requires it -- for
       // example, the rules for multiple choice need to refer to the multiple
       // choice interaction's customization arguments.
-      updateAnswerChoices: function(newAnswerChoices, callback) {
+      updateAnswerChoices: function(
+          newAnswerChoices, oldToNewListMapping, callback) {
         var oldAnswerChoices = angular.copy(_answerChoices);
         _answerChoices = newAnswerChoices;
 
+        // If the interaction is MultipleChoiceInput, update the answer groups
+        // to refer to the new answer options.
+        if (StateInteractionIdService.savedMemento === 'MultipleChoiceInput') {
+          // Save and update the answer groups for MultipleChoiceInput
+          // based on the information about the modification of answer choices
+          // from the schema-based-list editor.
+          // The information regarding the modifications of answer choices is
+          // present in oldToNewListMapping.
+
+          if (oldToNewListMapping !== undefined) {
+            var deletedIndexes = oldToNewListMapping.deletedIndexes;
+            var answerGroups = angular.copy(_answerGroups);
```

Comment:
I'm not sure I understand why you need to make a copy? It seems to me that it just makes things confusing because both the copy and the original are named similarly.

Reply:
I've changed the naming.

> Im not sure I understand why you need to make a copy?

Here we need the copy because we are updating `_answerGroups` in the iterations for iteration.
If we directly use `_answerGroups` for iterating. Then when the `_answerGroups` gets updated the code won't work as expected. I hope it's more clear now.